### PR TITLE
Login navigation overhaul

### PR DESCRIPTION
**Pull request by ChocolaterToba**

### CODE REVIEW

File: presentation/src/main/java/com/example/pinterbest/presentation/LoginFragment.kt [hunk 1]
```diff
@@ -52,14 +78,17 @@ class LoginFragment : Fragment() {
...
+                .navigate(R.id.registrationFragment, registrationArgs.build().toBundle())
         }
     }
 
     private fun initObservers(view: View) {
         viewModel.response.observe(viewLifecycleOwner) {
-            view.findNavController().navigate(R.id.homeFragment)
-            setUpBottomNavigationItem()
+            println(returnFragmentID == R.id.profileFragment)
+            println(view.findNavController().popBackStack(returnFragmentID, false))
```
Comment (102Rudy) [inline presentation/src/main/java/com/example/pinterbest/presentation/LoginFragment.kt]: Please, use `Log.d` instead

File: presentation/src/main/java/com/example/pinterbest/presentation/MainActivity.kt [hunk 2]
```diff
@@ -124,13 +95,14 @@ class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelecte
...
-    override fun onBackPressed() {
-        if (navHostFragment.childFragmentManager.backStackEntryCount > 0) {
-            navController.navigateUp()
-            setMenuChecked(navController.currentDestination?.id)
-        } else {
-            super.onBackPressed()
-        }
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putBundle("current_navcontroller", navController.saveState())
```
Comment (102Rudy) [inline presentation/src/main/java/com/example/pinterbest/presentation/MainActivity.kt]: let's extract `"current_navcontroller"` into constant

File: presentation/src/main/java/com/example/pinterbest/presentation/PinCreationFragment.kt [hunk 3]
```diff
@@ -61,7 +67,7 @@ class PinCreationFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        initObservers()
+        authObservers()
```
Comment (102Rudy) [inline presentation/src/main/java/com/example/pinterbest/presentation/PinCreationFragment.kt]: initAuthObservers?

File: presentation/src/main/java/com/example/pinterbest/presentation/ProfileFragment.kt [hunk 4]
```diff
@@ -73,7 +75,7 @@ class ProfileFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        initObservers()
+        registrationObservers()
```
Comment (102Rudy) [inline presentation/src/main/java/com/example/pinterbest/presentation/ProfileFragment.kt]: initRegistrationObservers

**Review by 102Rudy — APPROVED** [review]

File: presentation/src/main/java/com/example/pinterbest/presentation/LoginFragment.kt (hunk 1 above)
Comment (ChocolaterToba) [reply presentation/src/main/java/com/example/pinterbest/presentation/LoginFragment.kt]: This code was used exclusively for debug, I just forgot to delete it
Noted

**Pull request merged**